transport-native-unix-common/src/main/c/netty_unix_socket.c

```
@@ -188,17 +188,48 @@ static int socket_type(JNIEnv* env) {
     }
 }
 
+static void netty_unix_socket_optionHandleError(JNIEnv* env, int err, char* method) {
+    if (err == EBADF) {
+        netty_unix_errors_throwClosedChannelException(env);
+    } else {
+        netty_unix_errors_throwChannelExceptionErrorNo(env, method, err);
+    }
+}
+
+static void netty_unix_socket_setOptionHandleError(JNIEnv* env, int err) {
+    netty_unix_socket_optionHandleError(env, err, "setsockopt() failed: ");
+}
+
+static int netty_unix_socket_setOption0(jint fd, int level, int optname, const void* optval, socklen_t len) {
+    return setsockopt(fd, level, optname, optval, len);
+}
+
+int netty_unix_socket_setOption(JNIEnv* env, jint fd, int level, int optname, const void* optval, socklen_t len) {
+    int rc = netty_unix_socket_setOption0(fd, level, optname, optval, len);
+    if (rc < 0) {
+        netty_unix_socket_setOptionHandleError(env, errno);
+    }
+    return rc;
+}
+
 static jint _socket(JNIEnv* env, jclass clazz, int type) {
     int fd = nettyNonBlockingSocket(socketType, type, 0);
     if (fd == -1) {
         return -errno;
     } else if (socketType == AF_INET6) {
-        // Allow to listen /connect ipv4 and ipv6
+        // Try to allow listen /connect ipv4 and ipv6
         int optval = 0;
-        if (netty_unix_socket_setOption(env, fd, IPPROTO_IPV6, IPV6_V6ONLY, &optval, sizeof(optval)) < 0) {
-            // Something went wrong so close the fd and return here. setOption(...) itself throws the exception already.
-            close(fd);
-            return -1;
+        if (netty_unix_socket_setOption0(fd, IPPROTO_IPV6, IPV6_V6ONLY, &optval, sizeof(optval)) < 0) {
+            if (errno == EAFNOSUPPORT) {
+              // Something went wrong, but don't close fd and don't return -1.   At best we can do is log.
+              // TODO: bubble this up to an actual Logger.
+              fprintf(stderr, "setsockopt() failed: Unable to set IPV6_V6ONLY.  Continuing.\n");
```

Why do we want to suppress the error here? `socket_type` accounts for `EAFNOSUPPORT` so if we get it here it seems like a programming error. Can we just propagate the error via `netty_unix_socket_setOptionHandleError` in this case too?